Uptane docs

Abhijay007 opened the pull request.

Abhijay007 committed:
Chore : Removed Jekyll env and unnecessary files

Signed-off-by: Abhijay jain <Abhijay007j@gmail.com>

Abhijay007 committed:
Chore : initialized with docusaurus build env

Signed-off-by: Abhijay jain <Abhijay007j@gmail.com>

Abhijay007 committed:
Refactor: Updated configuration for Uptane

Signed-off-by: Abhijay jain <Abhijay007j@gmail.com>

Abhijay007 committed:
docs : Added initial GSoC blog entry

Signed-off-by: Abhijay jain <Abhijay007j@gmail.com>

tkfu commented:
First, this will need a rebase. Update master on your branch so that it's up to date with uptane/uptane.github.io:master first, then do a rebase and force push.

Second, I have some git hygiene issues with this PR, that should be fairly easy to fix:
* [7a28d95](https://github.com/Abhijay007/Abhijay007.github.io/pull/1/commits/7a28d9522af876ecfa459c1db76f243a9cd54c71)  and [498fa59](https://github.com/Abhijay007/Abhijay007.github.io/pull/1/commits/498fa595224f7ebe0c0373eb025ddb6b6c50e70a) should be squashed into one commit. This is just to keep the history clean; there's no need for the files you deleted to be added to the repo in the first place.
* [8236c3b](https://github.com/Abhijay007/Abhijay007.github.io/pull/1/commits/8236c3ba46b1424b7d688f7636c70e2fe68e432a) should be split into two commits: first a commit to rename/move the files, and then a second commit with the various formatting changes you've applied. This helps reviewers actually see the changes you've made. With the rename and edits in the same commit, difftool isn't able to tell that you made edits to the moved files, so it shows up as though you added a bunch of brand new files.

Once that's done, this branch can serve as a basis for adding the build pipeline.

tkfu commented:
However, I'll note that the build is currently failing because of a bunch of broken links:
```
λ yarn build
yarn run v1.22.10
$ docusaurus build
[INFO] [en] Creating an optimized production build...

✔ Client


✔ Server
  Compiled successfully in 2.17s


● Client █████████████████████████ cache (99%) shutdown IdleFileCachePlugin
 stored

● Server █████████████████████████ cache (99%) shutdown IdleFileCachePlugin
 serialize pack

[ERROR] Unable to build website for locale en.
[ERROR] Error: Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

It looks like some of the broken links we found appear in many pages of your site.
Maybe those broken links appear on all pages through your site layout?
We recommend that you check your theme configuration for such links (particularly, theme navbar and footer).
Frequent broken links are linking to:
- /docs/intro

Exhaustive list of all broken links found:

- On source page path = /404.html:
   -> linking to /docs/intro

- On source page path = /blog:
   -> linking to /docs/intro

- On source page path = /blog/archive:
   -> linking to /docs/intro

- On source page path = /blog/blog Index 0:
   -> linking to /docs/intro

- On source page path = /markdown-page:
   -> linking to /docs/intro

- On source page path = /docs/:
   -> linking to /papers/Uptane_meeting_agenda_62323.pdf
   -> linking to /papers/uptane-standard.1.1.0.html
   -> linking to /papers/uptane-standard.1.2.0.html
   -> linking to /papers/uptane-standard.2.0.0.html
   -> linking to /uptane-standard/2.1.0/uptane-standard.html
   -> linking to /deployment-considerations/index.html
   -> linking to /docs/intro

- On source page path = /docs/adoptions:
   -> linking to /docs/intro

- On source page path = /docs/alliance:
   -> linking to /docs/intro

- On source page path = /docs/attacks:
   -> linking to /docs/intro

- On source page path = /docs/audio:
   -> linking to /papers/23uptane_331_mackay.pdf
   -> linking to /papers/23uptane_331_ms.pdf
   -> linking to /papers/22uptane_yokohama_december_7.pdf
   -> linking to /papers/22uptane_october_22_presentationmain.pdf
   -> linking to /papers/Lightman_ISO_24089.pdf
   -> linking to /papers/22uptane_russell_UNECEWP.29.pdf
   -> linking to /papers/Over-the-Air-Updates-PACCAR.pdf
   -> linking to /papers/IIW21_session1.pdf
   -> linking to /papers/IIW21_session2.pdf
   -> linking to /papers/IW20_lightning_update.pdf
   -> linking to /papers/IW20_aftermarket.pdf
   -> linking to /papers/IW20_emergency_updates.pdf
   -> linking to /papers/IW20_poufs.pdf
   -> linking to /docs/intro

- On source page path = /docs/audits:
   -> linking to /docs/intro

- On source page path = /docs/contact:
   -> linking to uptane@googlegroups.com (resolved as: /docs/uptane@googlegroups.com)
   -> linking to /docs/intro

- On source page path = /docs/deployment/introduction:
   -> linking to /docs/intro

- On source page path = /docs/design:
   -> linking to /docs/intro

- On source page path = /docs/faq:
   -> linking to /docs/intro

- On source page path = /docs/governance:
   -> linking to /docs/intro

- On source page path = /docs/overview:
   -> linking to /docs/intro

- On source page path = /docs/participate:
   -> linking to /docs/intro

- On source page path = /docs/people:
   -> linking to /docs/intro

- On source page path = /docs/pouf:
   -> linking to /docs/intro

- On source page path = /docs/press:
   -> linking to /docs/intro

- On source page path = /docs/publications:
   -> linking to /docs/intro

- On source page path = /docs/reference_pouf:
   -> linking to /docs/intro

- On source page path = /docs/standardref:
   -> linking to papers/uptane-standard.2.1.0.html (resolved as: /docs/papers/uptane-standard.2.1.0.html)
   -> linking to /uptane-standard/2.1.0/uptane-standard.pdf
   -> linking to /deployment-considerations/changelog.html#210---2023-06-6 (resolved as: /deployment-considerations/changelog.html)
   -> linking to papers/uptane-standard.2.0.0.html (resolved as: /docs/papers/uptane-standard.2.0.0.html)
   -> linking to /deployment-considerations/changelog.html#200---2022-01-18 (resolved as: /deployment-considerations/changelog.html)
   -> linking to papers/uptane-standard.1.2.0.html (resolved as: /docs/papers/uptane-standard.1.2.0.html)
   -> linking to papers/uptane-standard.1.1.0.html (resolved as: /docs/papers/uptane-standard.1.1.0.html)
   -> linking to papers/uptane-standard.1.0.1.html (resolved as: /docs/papers/uptane-standard.1.0.1.html)
   -> linking to papers/uptane-standard.1.0.1.pdf (resolved as: /docs/papers/uptane-standard.1.0.1.pdf)
   -> linking to papers/ieee-isto-6100.1.0.0.uptane-standard.html (resolved as: /docs/papers/ieee-isto-6100.1.0.0.uptane-standard.html)
   -> linking to papers/V2.0.0_uptane_deploy.html (resolved as: /docs/papers/V2.0.0_uptane_deploy.html)
   -> linking to /docs/intro

- On source page path = /docs/timeline:
   -> linking to /docs/intro

- On source page path = /:
   -> linking to /docs/intro

    at throwError (/Users/jon/tdx/scratchpad/tkfu.github.io/node_modules/@docusaurus/logger/lib/index.js:76:11)
    at handleBrokenLinks (/Users/jon/tdx/scratchpad/tkfu.github.io/node_modules/@docusaurus/core/lib/server/brokenLinks.js:153:47)
    at async buildLocale (/Users/jon/tdx/scratchpad/tkfu.github.io/node_modules/@docusaurus/core/lib/commands/build.js:186:5)
    at async tryToBuildLocale (/Users/jon/tdx/scratchpad/tkfu.github.io/node_modules/@docusaurus/core/lib/commands/build.js:41:20)
    at async mapAsyncSequential (/Users/jon/tdx/scratchpad/tkfu.github.io/node_modules/@docusaurus/utils/lib/jsUtils.js:34:24)
    at async Command.build (/Users/jon/tdx/scratchpad/tkfu.github.io/node_modules/@docusaurus/core/lib/commands/build.js:76:21)
[INFO] Docusaurus version: 2.4.1
Node version: v18.12.0
error Command failed with exit code 1.
```

tkfu commented:
Once you've got the other issues with this branch sorted out, here's a workflow that can get you started. It only builds the site (no deployment), which is a good starting point. Once you've got passing builds, you can iterate and add a deployment step.

```
name: Build

on:
  push:
    branches: [ "master" ]
  pull_request:
    branches: [ "master" ]
  workflow_dispatch:


jobs:
  build:
    name: Build uptane website
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version: 18
          cache: yarn

      - name: Install dependencies
        run: yarn install --frozen-lockfile
      - name: Build website
        run: yarn build
```

tkfu commented:
Oh, and get rid of package-lock.json, and create yarn.lock. yarn is the recommended build tool for docusaurus.

Abhijay007 commented:
Thanks for the review, @tkfu! I will look into your suggestions and resolve them as soon as possible.

Abhijay007 commented:
> First, this will need a rebase. Update the master on your branch so that it's up to date with uptane/uptane.github.io:master first, then do a rebase and force push.

Hi @tkfu, I followed these steps for your first suggestion. I want you to cross-check it once in case I made any mistakes while updating the master on my branch and making sure it's up to date with uptane/uptane.github.io:master. Here's what I did:

First, I updated my local branch with uptane/uptane.github.io:master by fetching the latest changes from the upstream remote using this command:

```
git fetch upstream
```
Then, I checked out to my branch:

```
git checkout uptaneDocs
```

Next, I merged the changes from uptane/uptane.github.io:master into my local "uptaneDocs" branch:

```
git merge upstream/master
```

Now, my branch is up to date with uptane/uptane.github.io:master.

To perform the rebase, I used the following command:

```
git rebase uptaneDocs master
```

I received a message saying `Successfully rebased and updated refs/heads/master`, but I'm not entirely sure if I needed to use `git rebase master` instead of the command I used. When I tried that command, it resulted in many conflicts and errors. Could you please guide me on this? is `git rebase uptaneDocs master` the right command to use?

Also now I moved to step two which is squashing and splitting the commits

Abhijay007 committed:
Chore : resolved rebase conficts

Signed-off-by: Abhijay jain <Abhijay007j@gmail.com>

Abhijay007 committed:
Chore : Rename/moveed files

Signed-off-by: Abhijay jain <Abhijay007j@gmail.com>

Abhijay007 committed:
Chore : Apply various formatting changes

Signed-off-by: Abhijay jain <Abhijay007j@gmail.com>

Abhijay007 committed:
Chore : Temporarily fix broken build issues

Signed-off-by: Abhijay jain <Abhijay007j@gmail.com>

tkfu commented on _data/navbar.yml:
This seems to be a rebasing error--you've added the navbar from the jekyll site back.

tkfu commented on papers/V2.1.0_uptane_deploy.html:
Another rebase mistake, maybe? These papers have nothing to do with the package lock files.

tkfu commented on docs/participate.md:
Unrelated to build issues. Please keep your commits clean, so they can be properly reviewed.

tkfu commented on docusaurus.config.js:
```diff
@@ -21,7 +21,7 @@ const config = {
   projectName: 'uptane.github.io', // Usually your repo name.
 
-  onBrokenLinks: 'throw',
+  onBrokenLinks: 'ignore',
```
We absolutely should not be ignoring broken links.

Abhijay007 replied:
I added this for the temporary purpose, as I plan to add a 'Check Broken Links' workflow once we are done with the initial setup. as I thought that We will address broken links afterward, I also mentioned it in the commit message, but ya addressing them makes senses, I was just ignoring this for now as we might change the path for some folders as well in the future, then files pointing to those folders might get affected.

tkfu commented on .github/workflows/build-deploy.yml:
Let's save this for another PR, once the rest of the issues are taken care of.

Abhijay007 committed:
replaced package-lock.json with yarn.lock

Signed-off-by: Abhijay jain <Abhijay007j@gmail.com>

Abhijay007 committed:
chore : Fixed rebasing error

Signed-off-by: Abhijay jain <Abhijay007j@gmail.com>

Abhijay007 commented:
> However, I'll note that the build is currently failing because of a bunch of broken links:
> 

Hi @tkfu , I've managed to resolve some of the links, but I'm encountering an issue. It appears that most of the links mentioned in the broken build are not actually broken; they are pointing to the correct files, still while running `yarn build` it is showing that these links are broken. Can you suggest what I should do in this case?

````
[ERROR] Unable to build website for locale en.
[ERROR] Error: Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken links found:

- On source page path = /docs/:
   -> linking to papers/uptane-standard.1.1.0.html (resolved as: /docs/papers/uptane-standard.1.1.0.html)
   -> linking to papers/uptane-standard.1.2.0.html (resolved as: /docs/papers/uptane-standard.1.2.0.html)
   -> linking to papers/uptane-standard.2.0.0.html (resolved as: /docs/papers/uptane-standard.2.0.0.html)
   -> linking to /uptane-standard/2.1.0/uptane-standard.html

- On source page path = /docs/contact:
   -> linking to uptane@googlegroups.com (resolved as: /docs/uptane@googlegroups.com)

- On source page path = /docs/standardref:
   -> linking to papers/uptane-standard.2.1.0.html (resolved as: /docs/papers/uptane-standard.2.1.0.html)
   -> linking to /uptane-standard/2.1.0/uptane-standard.pdf
   -> linking to /deployment-considerations/changelog.html#210---2023-06-6 (resolved as: /deployment-considerations/changelog.html)
   -> linking to papers/uptane-standard.2.0.0.html (resolved as: /docs/papers/uptane-standard.2.0.0.html)
   -> linking to /deployment-considerations/changelog.html#200---2022-01-18 (resolved as: /deployment-considerations/changelog.html)
   -> linking to papers/uptane-standard.1.2.0.html (resolved as: /docs/papers/uptane-standard.1.2.0.html)
   -> linking to papers/uptane-standard.1.1.0.html (resolved as: /docs/papers/uptane-standard.1.1.0.html)
   -> linking to papers/uptane-standard.1.0.1.html (resolved as: /docs/papers/uptane-standard.1.0.1.html)
   -> linking to papers/uptane-standard.1.0.1.pdf (resolved as: /docs/papers/uptane-standard.1.0.1.pdf)
   -> linking to /docs/papers/ieee-isto-6100.1.0.0.uptane-standard.html
   -> linking to papers/V2.0.0_uptane_deploy.html (resolved as: /docs/papers/V2.0.0_uptane_deploy.html)
   -> linking to /docs/papers/V2.0.0_uptane_deploy.html
   
   ````

hexsecs commented:
Abhijay,  suggest cancelling this pull request and make a new one with the latest content. You have made many changes and updates since this pull request.

Abhijay007 commented:
> Abhijay, suggest cancelling this pull request and make a new one with the latest content. You have made many changes and updates since this pull request.

sure I added new branch here : https://github.com/Abhijay007/Abhijay007.github.io/tree/uptaneDemo
